server/service/orbit.go

```diff
@@ -890,6 +890,37 @@ func (svc *Service) SaveHostSoftwareInstallResult(ctx context.Context, result *f
 
 	// always use the authenticated host's ID as host_id
 	result.HostID = host.ID
-	err := svc.ds.SetHostSoftwareInstallResult(ctx, result)
-	return ctxerr.Wrap(ctx, err, "save host software installation result")
+	if err := svc.ds.SetHostSoftwareInstallResult(ctx, result); err != nil {
+		return ctxerr.Wrap(ctx, err, "save host software installation result")
+	}
+
+	if status := result.Status(); status != fleet.SoftwareInstallerPending {
```

Comment:
Should never happen, but just in case - not creating the (past) activity if the status is pending as it will still show up in the "upcoming" activities.